src/aspire/source/image.py

```diff
@@ -597,16 +596,68 @@ def save(
             overwrite=overwrite,
         )
 
-    def save_metadata(
-        self, starfile_filepath, new_mrcs=True, batch_size=512, save_mode=None
+    def _populate_common_metadata(
+        self,
+        df,
+        local_cols,
+        starfile_filepath,
+        batch_size,
+        save_mode,
     ):
+        """
+        Populate metadata columns common to all `ImageSource` subclasses.
+        """
+        # Create a new column that we will be populating in the loop below
+        df["_rlnImageName"] = ""
+
+        if save_mode == "single":
```

Comment:
Chris, can you double check if there is any difference between the code in the branches other than the `for` loop?  Looking quickly, it would appear to be the same code. If it is, I think we can just run the loop for a single iteration and drop half this code...

Reply:
I'll see if i can refactor this

Reply:
I think we can leave this for #745